src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,30 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object rectangle) {
+        if (rectangle == this) {
+            return true;
+        }
+        if (rectangle == null) {
+            return false;
+        }
+        Rectangle castedRectangle = (Rectangle) rectangle;
+        if (!castedRectangle.getClass().equals(Rectangle.class)) {
+            return false;
+        }
+        return (color == null || color.equals(castedRectangle.color))
+                && (width == null || width.equals(castedRectangle.width))
+                && (length == null || length.equals(castedRectangle.length));
```

Comment:
let's follow the following logical flow for each field:

return true, 
if both `field` and `rectangle.field` are null 
OR if `field` is not null and equals `rectangle.field`